spring-session-core/src/main/java/org/springframework/session/PrincipalNameIndexResolver.java

```diff
@@ -34,6 +34,8 @@
 
 	private static final SpelExpressionParser parser = new SpelExpressionParser();
 
+	private static final Expression expression = parser.parseExpression("authentication?.name");
```

Comment:
I think we should also get rid of `parser` as member variable and simply instantiate it here as it won't be reused.